gnc/sub8_trajectory_generator/src/tgen_manager.cc

```diff
@@ -0,0 +1,257 @@
+/**
+ * Author: Patrick Emami
+ * Date: 9/29/15
+ *
+ */
+
+#include "tgen_manager.h"
+#include "tgen_common.h"
+#include "sub8_state_space.h"
+#include "ompl/base/PlannerStatus.h"
+#include "ompl/base/goals/GoalState.h"
+#include "ompl/geometric/planners/rrt/RRTstar.h"
+#include "ompl/geometric/planners/rrt/RRTConnect.h"
+#include "ompl/geometric/PathGeometric.h"
+#include "ompl/base/spaces/RealVectorBounds.h"
+#include <boost/filesystem.hpp>
+#include <ros/console.h>
+#include <ros/package.h>
+#include <Eigen/Dense>
+#include <Eigen/Geometry>
+#include <cmath>
+
+// File IO to save solution to file
+#if 0
+#include <iostream>
+#include <fstream>
+#endif
+
+using sub8::trajectory_generator::TGenManager;
+using sub8::trajectory_generator::TGenMsgs;
+using sub8::trajectory_generator::SpaceInformationGeneratorPtr;
+using sub8::trajectory_generator::Sub8StateSpace;
+using ompl::base::Planner;
+using ompl::base::PlannerStatus;
+using ompl::base::GoalState;
+using ompl::base::ProblemDefinitionPtr;
+using ompl::base::RealVectorBounds;
+using ompl::geometric::PathGeometric;
+using ompl::geometric::RRTConnect;
+using ompl::geometric::RRTstar;
+
+namespace fs = ::boost::filesystem;
+
+TGenManager::TGenManager(AlarmBroadcasterPtr&& ab) : _alarm_broadcaster(ab) {
+  int planner_type;
+  int range;
+
+  _pdef = nullptr;  // Problem definition hasn't been set yet
+  SpaceInformationGeneratorPtr ss_gen(new SpaceInformationGenerator());
+
+  _sub8_si = ss_gen->generate();
+
+  ros::param::get("planner_type", planner_type);
+  ros::param::get("range", range);
+
+  switch (planner_type) {
+    case 1:
+      _sub8_planner =
+          boost::shared_ptr<Planner>(new ompl::geometric::RRTstar(_sub8_si));
+      ROS_INFO("Using OMPL's RRTstar path-planner");
+      // set the maximum distance the planner can extend itself per "step" in
+      // meters
+      _sub8_planner->as<ompl::geometric::RRTstar>()->setRange(range);
+      break;
+    default:
+      _sub8_planner =
+          boost::shared_ptr<Planner>(new ompl::geometric::RRTConnect(_sub8_si));
+      ROS_INFO("Using OMPL's RRTConnect path-planner");
+      // set the maximum distance the planner can extend itself per "step" in
+      // meters
+      _sub8_planner->as<ompl::geometric::RRTConnect>()->setRange(range);
+  }
+
+  // initialize alarms
+  std::string alarms_dir;
+  ros::param::get("alarms_dir", alarms_dir);
+  std::string pkg_path = ros::package::getPath("sub8_trajectory_generator");
+  char file_sep = '/';
+#ifdef _WIN32
```

Comment:
C++ alarms allow the user to specify alarms with JSON, by placing them in an "alarms" directory. I'm gonna move most of this code to the alarm helper class though in a new commit.